bndtools.builder/src/org/bndtools/builder/classpath/BndContainerInitializer.java

```diff
@@ -303,16 +311,50 @@ static boolean isProjectContainer(Container container) {
         return attrs.toArray(new IClasspathAttribute[0]);
     }
 
-    static IAccessRule[] calculateRepoBundleAccessRules(Container c) {
+    static IAccessRule[] calculateRepoBundleAccessRules(Container c, Project model) {
+        LinkedList<IAccessRule> tmp = null;
         String packageList = c.getAttributes().get("packages");
         if (packageList != null) {
-            List<IAccessRule> tmp = new LinkedList<IAccessRule>();
+            tmp = new LinkedList<IAccessRule>();
             StringTokenizer tokenizer = new StringTokenizer(packageList, ",");
             while (tokenizer.hasMoreTokens()) {
                 String token = tokenizer.nextToken();
                 String pathStr = token.replace('.', '/') + "/*";
                 tmp.add(JavaCore.newAccessRule(new Path(pathStr), IAccessRule.K_ACCESSIBLE));
             }
+        } else {
+            // If there are any exports from a version=file reference, provide access rules
+            // to allow these packages to be used/seen by users of the "this" project.
+            // e.g.
+            // Project A generates A.jar with java package A
+            // Project B has a buildpath version=file reference to A.jar and exports package A
+            // Project C has a buildpath version=snapshot|latest reference to B and uses package A.
+            // The access rules calculated by this section permit Project C to
+            // use packages from Project A in eclipse.
+            Packages exportPkgs = model.getExports();
+            ZipFile zip = null;
+            try {
+                zip = new ZipFile(c.getFile());
```

Comment:
Instead of using ZipFile, I think it would be better to use aQute.bnd.osgi.Jar. You can then use the Jar object to interrogate the directories or package (which ever is the correct choice here).